tests/testthat/test-shinytest2-tm_file_viewer.R

```diff
@@ -1,11 +1,62 @@
-test_that("e2e: tm_file_viewer initializes without errors and shows encodings", {
+app_driver_tm_file_viewer <- function() {
+  TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = tm_file_viewer(
+      label = "File Viewer Module",
+      input_path = list(
+        folder = system.file("sample_files", package = "teal.modules.general"),
+        png = system.file("sample_files/sample_file.png", package = "teal.modules.general"),
+        txt = system.file("sample_files/sample_file.txt", package = "teal.modules.general"),
+        url = "https://fda.gov/files/drugs/published/Portable-Document-Format-Specifications.pdf"
+      )
+    ),
+    timeout = 3000
+  )
+}
+
+test_that("e2e: tm_file_viewer initializes without errors and shows files tree specified in input_path argument", {
   skip_if_too_deep(5)
   app_driver <- app_driver_tm_file_viewer()
 
   app_driver$expect_no_shiny_error()
 
   # encoding are visible
   testthat::expect_true(app_driver$is_visible(selector = app_driver$active_module_element("tree")))
+  list_files <- app_driver$get_html_rvest(selector = app_driver$active_module_element("tree")) %>%
+    rvest::html_nodes("a") %>%
+    rvest::html_text()
+  testthat::expect_setequal(
+    list_files,
+    c("folder", "png", "txt", "url")
+  )
+
+
+  app_driver$stop()
+})
+
+test_that("e2e: tm_file_viewer shows selected image file", {
+  skip_if_too_deep(5)
+  app_driver <- app_driver_tm_file_viewer()
+
+  app_driver$click(selector = "[id= '4_anchor']")
+  testthat::expect_true(app_driver$is_visible(selector = app_driver$active_module_element("output")))
```

Comment:
Is `$is_visible` good enough to say that image is there? I don't think so. In theory there could be an empty div with `id="output"` and every test here will pass.

Reply:
If the link or path to a PNG or text file is incorrect or nonexistent,  then is warning
`"In tm_file_viewer(label = 'File Viewer Module', input_path = list(folder = system.file('sample_files', package = 'your_package_name'))): Non-existent file or URL path. Please provide valid paths for:"`
The expect_no_validation_error()  from tTealAppDriver  will test this. (adding this)

However, there is still a limitation when dealing with external URLs because their existence cannot be directly verified to check this I can add test or the presence of` pre` tags for displaying text and` img `tags for displaying images within the HTML output. WDYT ?

Currently, we are not utilizing snapshot testing.

Reply:
added test to checks tags  [3756dfe](https://github.com/insightsengineering/teal.modules.general/pull/723/commits/3756dfe290f5701b11e20625581c69ba30370f43)